apps/backend/src/apikeys/apikey.service.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  ForbiddenException,
+  Injectable,
+  NotFoundException
+} from '@nestjs/common';
+import {InjectModel} from '@nestjs/sequelize';
+import {hash} from 'bcryptjs';
+import jwt from 'jsonwebtoken';
+import {CreateApiKeyDto} from '../apikeys/dto/create-apikey.dto';
+import {ConfigService} from '../config/config.service';
+import {User} from '../users/user.model';
+import {UsersService} from '../users/users.service';
+import {ApiKey} from './apikey.model';
+import {APIKeyDto} from './dto/apikey.dto';
+import {UpdateAPIKeyDto} from './dto/update-apikey.dto';
+
+@Injectable()
+export class ApiKeyService {
+  constructor(
+    @InjectModel(ApiKey)
+    private readonly apiKeyModel: typeof ApiKey,
+    private readonly configService: ConfigService,
+    private readonly usersService: UsersService
+  ) {}
+
+  async count(): Promise<number> {
+    return this.apiKeyModel.count();
+  }
+
+  async create(
+    user: User,
+    createApiKeyDto: CreateApiKeyDto
+  ): Promise<{id: string; name: string; apiKey: string}> {
+    const JWTSecret = this.configService.get('JWT_SECRET');
```

Comment:
instead of using this JWT_SECRET i think we just need to have something in the config service or app module that is required (app will fail if not set) for API_KEYS_SECRET or whatever

Reply:
Instead of `API_KEY_SECRET`, I'd argue that we just use `JWT_SECRET` for API keys and just generate `JWT_SECRET` as a random string on startup.

Reply:
I think that we should just stick with requiring `JWT_SECRET`. If we generated a new random string on startup, it would cause all existing API keys to break. (The app won't start without a JWT_SECRET anyways)

![image](https://user-images.githubusercontent.com/66680985/126174325-ca21455c-3293-4448-a6a0-3f6138e04cf5.png)

Reply:
What existing API keys? This feature hasn't been released yet

Reply:
The issue here is not whether the app starts or not, it is just that in general reusing secrets for multiple purposes is generally bad. This is made even worse by JWT's since the only way to expire a users session is to change the secret for all users, meaning expiring a single session would currently also expire all API keys. Having a different secret for API keys would be better. With that said, we don't really have a method of notifying users that they need to rerun `setup-docker-secrets.sh` to generate a new key, so my initial suggestion was to repurpose `JWT_SECRET` to be used for API keys and generate a new random secret for users on startup. With that said, maybe we just go for it and instead add a `API_KEY_SECRET` and have the feature only enable when `API_KEY_SECRET` is set, or just throw an error on startup when `API_KEY_SECRET` isn't set just like we do for JWT secrets.

Reply:
I've moved it to now check for `API_KEY_SECRET` and throw a ForbiddenException if not set on API-key related actions.